hal_st/cortex/InterruptCortex.cpp

```diff
@@ -59,49 +57,56 @@ namespace hal
     InterruptHandler::InterruptHandler(InterruptHandler&& other)
         : irq(other.irq)
     {
-        InterruptTable::Instance().TakeOverHandler(irq, *this, other);
-        other.irq = invalidIrq;
+        if (irq)
+            InterruptTable::Instance().TakeOverHandler(*irq, *this, other);
+
+        other.irq = infra::none;
     }
 
     InterruptHandler& InterruptHandler::operator=(InterruptHandler&& other)
     {
-        if (irq != invalidIrq)
-            InterruptTable::Instance().DeregisterHandler(irq, *this);
+        if (irq)
+            InterruptTable::Instance().DeregisterHandler(*irq, *this);
 
         irq = other.irq;
 
-        InterruptTable::Instance().TakeOverHandler(irq, *this, other);
-        other.irq = invalidIrq;
+        if (irq)
+            InterruptTable::Instance().TakeOverHandler(*irq, *this, other);
+
+        other.irq = infra::none;
 
         return *this;
     }
 
     InterruptHandler::~InterruptHandler()
     {
-        if (irq != invalidIrq)
-            InterruptTable::Instance().DeregisterHandler(irq, *this);
+        if (irq)
+            InterruptTable::Instance().DeregisterHandler(*irq, *this);
     }
 
     void InterruptHandler::Register(IRQn_Type irq, InterruptPriority priority)
     {
         this->irq = irq;
-        InterruptTable::Instance().RegisterHandler(this->irq, *this, priority);
+        InterruptTable::Instance().RegisterHandler(*this->irq, *this, priority);
     }
 
     void InterruptHandler::Unregister()
     {
-        InterruptTable::Instance().DeregisterHandler(irq, *this);
-        irq = invalidIrq;
+        if (irq)
+            InterruptTable::Instance().DeregisterHandler(*irq, *this);
+
+        irq = infra::none;
     }
 
-    IRQn_Type InterruptHandler::Irq() const
+    infra::Optional<IRQn_Type> InterruptHandler::Irq() const
     {
         return irq;
     }
 
     void InterruptHandler::ClearPending()
     {
-        NVIC_ClearPendingIRQ(irq);
+        if (irq)
```

Comment:
This if-statement should not be necessary. ClearPending should only be called on interrupt handlers that are registered